### PR TITLE
Fixes runtime issue #1739

### LINKED ISSUE #1739: Assertion failed 'varDsc->lvType == lcl->gtType || varDsc->lvType == TYP_BLK' in 'System.Reflection.CustomAttribute:CreateCaObject(System.Reflection.RuntimeModule,System.RuntimeType,System.IRuntimeMethodInfo,byref,long,byref):System.Object'
Windows x64 Checked (possibly other configurations as well),

Failure in Loader\binding\tracing\BinderTracingTest\BinderTracingTest.cmd with COMPlus_JitStress=1 or 2, and with tiering enabled. With `complus_TieredCompilation=0`, it doesn't repro. Also passes with `COMPlus_JITMinOpts=1`.

https://dev.azure.com/dnceng/public/_build/results?buildId=482702&view=ms.vss-test-web.build-test-results-tab&runId=15325494&resultId=110322&paneView=debug

```
Assert failure(PID 18164 [0x000046f4], Thread: 17768 [0x4568]): Assertion failed 'varDsc->lvType == lcl->gtType || varDsc->lvType == TYP_BLK' in 'System.Reflection.CustomAttribute:CreateCaObject(System.Reflection.RuntimeModule,System.RuntimeType,System.IRuntimeMethodInfo,byref,long,byref):System.Object' (IL size 46)

    File: F:\gh\runtime2\src\coreclr\src\jit\lclvars.cpp Line: 7363
    Image: F:\gh\runtime2\artifacts\tests\coreclr\Windows_NT.x64.Checked\Tests\Core_Root\CoreRun.exe
```

### PR DESCRIPTION
**Pull request by briansull**
Added additional check for Jit Stress mode STRESS_LCL_FLDS
This stress mode requires that the types exactly match

Fixes #1739

### CODE REVIEW

**Comment by briansull** [comment]
@dotnet/jit-contrib PTAL

File: src/coreclr/src/jit/lclvars.cpp [hunk 1]
```diff
@@ -7339,6 +7339,14 @@ Compiler::fgWalkResult Compiler::lvaStressLclFldCB(GenTree** pTree, fgWalkData*
             return WALK_SKIP_SUBTREES;
         }
 
+        // The assert in the second pass below, requires that these types match, or we have a TYP_BLK
+        //
+        if ((varDsc->lvType != lcl->gtType) && varDsc->lvType != TYP_BLK)
```
Comment (AndyAyersMS) [inline src/coreclr/src/jit/lclvars.cpp]: Nit: parens on the second part of the conditional...

**Review by AndyAyersMS — APPROVED** [review]
Changes LGTM.

I wonder if this stress mode provides a lot of value.

File: src/coreclr/src/jit/lclvars.cpp (hunk 1 above)
Comment (briansull) [reply src/coreclr/src/jit/lclvars.cpp]: Updated with extra parens in both places

**Comment by briansull** [comment]
The only test failure is due to the current issue with EvenPipe timeouts due to a deadlock:
Fix for that issue is being worked on here #1794

**Pull request merged**